server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -232,7 +237,7 @@ private List<StoreFileMetadata> getFiles(CheckpointInfoResponse checkpointInfo)
         return missingFiles;
     }
 
-    private boolean validateLocalChecksum(StoreFileMetadata file) {
+    private boolean validateLocalChecksum(StoreFileMetadata file) throws IOException {
```

Comment:
is throws needed here if we are catching IOException in the method ?

Reply:
It is needed as we are deleting the file.

Reply:
Got it